### PR TITLE
include: add EOVERFLOW status code mapping

### PR DESCRIPTION
**Pull request by RaisinTen**
Refs: https://github.com/nodejs/node/pull/38159#discussion_r610288214

### CODE REVIEW

**Review by richardlau — CHANGES_REQUESTED** [review]
You'll need to update `include/uv/errno.h` too.

File: include/uv/errno.h [hunk 1]
```diff
@@ -302,6 +302,12 @@
 # define UV__ENOTSUP (-4049)
 #endif
 
+#if defined(EOVERFLOW) && !defined(_WIN32)
+# define UV__OVERFLOW UV__ERR(EOVERFLOW)
+#else
+# define UV__OVERFLOW (-4026)
+#endif
+
```
Comment (richardlau) [inline include/uv/errno.h:310]: I think the entries in this file are ordered in descending assigned value (i.e. `-4026`) comes last. That would make it easier to see what the value the next added entry should take.

Comment (RaisinTen) [reply include/uv/errno.h:310]: These are also in alphabetical order. Should we update all the error codes underneath?

Comment (richardlau) [reply include/uv/errno.h:310]: `UV__OVERFLOW` should be `UV__EOVERFLOW`.

Comment (richardlau) [reply include/uv/errno.h]: > These are also in alphabetical order. Should we update all the error codes underneath?

? They're not in alphabetical order.

Comment (RaisinTen) [reply include/uv/errno.h]: Just noticed it, moved it to the bottom. :)

**Review by richardlau — APPROVED** [review]

**Comment by richardlau** [comment]
CI: https://ci.nodejs.org/job/libuv-test-commit/2144/

File: docs/src/errors.rst [hunk 2]
```diff
@@ -251,6 +251,10 @@ Error constants
 
     operation not supported on socket
 
+.. c:macro:: UV_EOVERFLOW
+
+    value too large for defined data types
```
Comment (cjihrig) [inline docs/src/errors.rst:256]:
```suggestion
    value too large for defined data type
```

**Review by cjihrig — APPROVED** [review]

File: docs/src/errors.rst (hunk 2 above)
Comment (RaisinTen) [reply docs/src/errors.rst]: @cjihrig Done. :)

**Review by vtjnash — APPROVED** [review]

**Pull request merged**